pkg/skaffold/docker/remote.go

```diff
@@ -62,21 +46,41 @@ func RemoteDigest(identifier string) (string, error) {
 }
 
 // RetrieveRemoteConfig retrieves the remote config file for an image
-func RetrieveRemoteConfig(identifier string) (*v1.ConfigFile, error) {
-	img, err := remoteImage(identifier)
+func RetrieveRemoteConfig(identifier string, insecureRegistries map[string]bool) (*v1.ConfigFile, error) {
+	img, err := remoteImage(identifier, insecureRegistries)
 	if err != nil {
 		return nil, errors.Wrap(err, "getting image")
 	}
 
 	return img.ConfigFile()
 }
 
-func remoteImage(identifier string) (v1.Image, error) {
-	ref, err := name.ParseReference(identifier, name.WeakValidation)
+func remoteImage(identifier string, insecureRegistries map[string]bool) (v1.Image, error) {
+	ref, err := name.ParseReference(identifier)
 	if err != nil {
 		return nil, errors.Wrap(err, "parsing initial ref")
 	}
 
+	if isInsecure(ref.Context().Registry.Name(), insecureRegistries) {
```

Comment:
As far as I understand, the `insecureRegistries` is only required here.
Would an package `init` which retrieves the insecure registries from the config file be a valid approach? That would have the advantage to not scatter the configuration over large portions of the codebase.

Reply:
yeah we're thinking along the lines of creating a `RunContext` that could be passed down from the runner rather.